tests/testBasicCameraFactory.py

```diff
@@ -0,0 +1,278 @@
+            yFocalList = []
+            for xx in range(0, 400, 100):
+                for yy in range(0, 400, 100):
+                    xPix_control.append(xx)
+                    yPix_control.append(yy)
+
+                    # Note: because of the demand that the x-axis in pixel coordinates
+                    # be along the direction of readout, the pixel x and y axes are rotated
+                    # 90 degrees with respect to the focal plane x and y axes
+                    xxFocalUnRotated = (0.5*(nx-1)-yy)*mmPerPixel
+                    yyFocalUnRotated = (xx-0.5*(ny-1))*mmPerPixel
+
+                    xxFocal = xCenter + xxFocalUnRotated*np.cos(yaw) - yyFocalUnRotated*np.sin(yaw)
+                    yyFocal = yCenter + xxFocalUnRotated*np.sin(yaw) + yyFocalUnRotated*np.cos(yaw)
+
+                    xFocalList.append(xxFocal)
+                    yFocalList.append(yyFocal)
+
+            for xxF, yyF, xxPcontrol, yyPcontrol in \
+            zip(xFocalList, yFocalList, xPix_control, yPix_control):
+
+                focalPoint = camera.makeCameraPoint(afwGeom.Point2D(xxF, yyF), FOCAL_PLANE)
+                pixelPoint = camera.transform(focalPoint, pixelSystem).getPoint()
+
+                self.assertAlmostEqual(float(xxPcontrol), pixelPoint.getX(), 9)
+                self.assertAlmostEqual(float(yyPcontrol), pixelPoint.getY(), 9)
+
+
+    def testPupilCoords(self):
+        """
+        Test that cameras generated with BasicCameraFactory convert from pupil
+        coordinates to pixel coordinates as expected
+        """
+        layoutFile = os.path.join(self.cameraDataDir, 'testFocalPlaneLayout.txt')
+
+        def getIdFromName(name):
+            return int(name[-2:])
+
+        np.random.seed(435)
+
+        detNameList = ['Det00', 'Det01', 'Det02']
+        yawList = np.radians([20.0, 10.0, 30.0])
+        xCenterList = [0.0, 0.0, -9.0]
+        yCenterList = [0.0, 9.0, 0.0]
+        nxList = [400, 300, 300]
+        nyList = [400, 200, 300]
+        mmPerPixelList = [2.0e-3, 1.0e-3, 3.0e-3]
+
+        for iteration in range(3):
+
+            # try random radial transforms between pupil and pixel coordinates
+            radial_coeffs = np.random.random_sample(5)
+            radial_coeffs[0] = 0.0
+
+            factory = BasicCameraFactory(detectorLayoutFile=layoutFile,
+                                         detectorIdFromAbbrevName=getIdFromName,
+                                         radialTransform = radial_coeffs,
+                                         detTypeMap = {'science':SCIENCE,
+                                                       'focus':FOCUS,
+                                                       'guider':GUIDER,
+                                                       'wave':WAVEFRONT})
+
+
+            camera = factory.makeCamera()
+
+            for detName, yaw, xCenter, yCenter, nx, ny, mmPerPixel in \
+            zip(detNameList, yawList, xCenterList, yCenterList, \
```

Comment:
can you please indent this so that it doesn't look like a completely new statement lined up with the `for`. `flake8` will complain.